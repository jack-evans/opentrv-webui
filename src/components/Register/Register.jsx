import React, { Component } from 'react'
import { Form, TextInput } from 'carbon-components-react'

class Register extends Component {
  constructor (props) {
    super(props)
    this.state = {
      name: '',
      email: '',
      phone: '',
      password: '',
      address: {
        number: 0,
        road: '',
        city: '',
        county: '',
        country: '',
        postcode: ''
      }
    }
  }

  render () {
    return (
      <div className='Register'>
        <h2 className='Register__title'>Title</h2>
        <Form className='Register__form'>
          <TextInput
            id='register-first-name'
            labelText='First'
            placeholder='Enter first name'
          />
          <TextInput
            id='register-last-name'
            labelText='Last'
            placeholder='Enter last name'
          />
        </Form>
      </div>
    )
  }
}

export default Register
