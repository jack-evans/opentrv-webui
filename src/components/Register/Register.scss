
.Register {
  display: inline-block;
  justify-content: center;

  .Register__title {
    padding: 50px 0 20px 0;
  }

  .Register__form {
    width: 500px;
  }
}