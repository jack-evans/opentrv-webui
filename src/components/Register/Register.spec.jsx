import React from 'react'
import Register from './Register'
import Enzyme, { shallow, mount } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

Enzyme.configure({ adapter: new Adapter() })

describe('Register.jsx', () => {
  it('renders without crashing', () => {
    expect(shallow(<Register />)).toMatchSnapshot()
  })

  describe('when the user edits the name field', () => {
    let wrapper
    const name = 'John Doe'

    beforeEach(() => {
      wrapper = mount(<Register />)
      wrapper.find('input').at(0).simulate('change', { target: { value: name, name: 'name' } })
    })

    afterEach(() => {
      wrapper.unmount()
    })

    it('changes the name property in the user state', () => {
      expect(wrapper.state().user.name).toEqual(name)
    })
  })

  describe('when the user edits the email field', () => {
    let wrapper
    const email = 'john.doe@example.com'

    beforeEach(() => {
      wrapper = mount(<Register />)
      wrapper.find('input').at(1).simulate('change', { target: { value: email, name: 'email' } })
    })

    afterEach(() => {
      wrapper.unmount()
    })

    it('changes the email property in the user state', () => {
      expect(wrapper.state().user.email).toEqual(email)
    })
  })

  describe('when the user edits the Line 1 address input', () => {
    let wrapper
    const firstLine = '123 example street'

    beforeEach(() => {
      wrapper = mount(<Register />)
      wrapper.find('input').at(2).simulate('change', { target: { value: firstLine, name: 'address->firstLine' } })
    })

    afterEach(() => {
      wrapper.unmount()
    })

    it('changes the address.firstLine property in the user state', () => {
      expect(wrapper.state().user.address.firstLine).toEqual(firstLine)
    })
  })

  describe('when the user edits the Line 2 address input', () => {
    let wrapper
    const secondLine = 'example town'

    beforeEach(() => {
      wrapper = mount(<Register />)
      wrapper.find('input').at(3).simulate('change', { target: { value: secondLine, name: 'address->secondLine' } })
    })

    afterEach(() => {
      wrapper.unmount()
    })

    it('changes the address.secondLine property in the user state', () => {
      expect(wrapper.state().user.address.secondLine).toEqual(secondLine)
    })
  })

  describe('when the user edits the city address input', () => {
    let wrapper
    const city = 'example city'

    beforeEach(() => {
      wrapper = mount(<Register />)
      wrapper.find('input').at(4).simulate('change', { target: { value: city, name: 'address->city' } })
    })

    afterEach(() => {
      wrapper.unmount()
    })

    it('changes the address.city property in the user state', () => {
      expect(wrapper.state().user.address.city).toEqual(city)
    })
  })

  describe('when the user edits the county address input', () => {
    let wrapper
    const county = 'county'

    beforeEach(() => {
      wrapper = mount(<Register />)
      wrapper.find('input').at(5).simulate('change', { target: { value: county, name: 'address->county' } })
    })

    afterEach(() => {
      wrapper.unmount()
    })

    it('changes the address.county property in the user state', () => {
      expect(wrapper.state().user.address.county).toEqual(county)
    })
  })

  describe('when the user edits the country address input', () => {
    let wrapper
    const country = 'uk'

    beforeEach(() => {
      wrapper = mount(<Register />)
      wrapper.find('input').at(6).simulate('change', { target: { value: country, name: 'address->country' } })
    })

    afterEach(() => {
      wrapper.unmount()
    })

    it('changes the address.firstLine property in the user state', () => {
      expect(wrapper.state().user.address.country).toEqual(country)
    })
  })

  describe('when the user edits the postcode address input', () => {
    let wrapper
    const postcode = 'ab12cd'

    beforeEach(() => {
      wrapper = mount(<Register />)
      wrapper.find('input').at(7).simulate('change', { target: { value: postcode, name: 'address->postcode' } })
    })

    afterEach(() => {
      wrapper.unmount()
    })

    it('changes the address.postcode property in the user state', () => {
      expect(wrapper.state().user.address.postcode).toEqual(postcode)
    })
  })

  describe('when the user edits the password input', () => {
    let wrapper
    const password = 'password'

    beforeEach(() => {
      wrapper = mount(<Register />)
      wrapper.find('input').at(8).simulate('change', { target: { value: password, name: 'password' } })
    })

    afterEach(() => {
      wrapper.unmount()
    })

    it('changes the address.firstLine property in the user state', () => {
      expect(wrapper.state().user.password).toEqual(password)
    })
  })

  describe('when the user edits the confirm password input', () => {
    let wrapper
    const confirmPass = 'password'

    beforeEach(() => {
      wrapper = mount(<Register />)
      wrapper.find('input').at(9).simulate('change', { target: { value: confirmPass, name: 'password-confirm' } })
    })

    afterEach(() => {
      wrapper.unmount()
    })

    it('changes the address.firstLine property in the user state', () => {
      expect(wrapper.instance().confirmPass).toEqual(confirmPass)
    })
  })

  describe('when the user fills in all the required fields', () => {
    let wrapper
    const arrayOfInputs = [
      {
        value: 'John Doe',
        name: 'name'
      }, {
        value: 'john.doe@example.com',
        name: 'email'
      }, {
        value: '123 example street',
        name: 'address->firstLine'
      }, {
        value: 'exampleCounty',
        name: 'address->county'
      }, {
        value: 'ab12cd',
        name: 'address->postcode'
      }, {
        value: 'Passw0rd',
        name: 'password'
      }, {
        value: 'Passw0rd',
        name: 'password-confirm'
      }
    ]

    beforeEach(() => {
      wrapper = mount(<Register />)
      for (let i = 0; i < arrayOfInputs.length; i++) {
        wrapper.find('input').at(i).simulate('change', { target: { value: arrayOfInputs[i].value, name: arrayOfInputs[i].name } })
      }
    })

    afterEach(() => {
      wrapper.unmount()
    })

    it('sets the isDisabled state to false', () => {
      expect(wrapper.state().isDisabled).toEqual(false)
    })
  })
})
