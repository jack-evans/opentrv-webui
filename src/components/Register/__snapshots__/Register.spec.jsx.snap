// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register.jsx renders without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Register />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2
          className="Register__title"
        >
          Title
        </h2>,
        <Form
          className="Register__form"
        >
          <TextInput
            className="bx--text__input"
            disabled={false}
            id="register-first-name"
            invalid={false}
            invalidText=""
            labelText="First"
            onChange={[Function]}
            onClick={[Function]}
            placeholder="Enter first name"
            type="text"
          />
          <TextInput
            className="bx--text__input"
            disabled={false}
            id="register-last-name"
            invalid={false}
            invalidText=""
            labelText="Last"
            onChange={[Function]}
            onClick={[Function]}
            placeholder="Enter last name"
            type="text"
          />
        </Form>,
      ],
      "className": "Register",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Title",
          "className": "Register__title",
        },
        "ref": null,
        "rendered": "Title",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <TextInput
              className="bx--text__input"
              disabled={false}
              id="register-first-name"
              invalid={false}
              invalidText=""
              labelText="First"
              onChange={[Function]}
              onClick={[Function]}
              placeholder="Enter first name"
              type="text"
            />,
            <TextInput
              className="bx--text__input"
              disabled={false}
              id="register-last-name"
              invalid={false}
              invalidText=""
              labelText="Last"
              onChange={[Function]}
              onClick={[Function]}
              placeholder="Enter last name"
              type="text"
            />,
          ],
          "className": "Register__form",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "className": "bx--text__input",
              "disabled": false,
              "id": "register-first-name",
              "invalid": false,
              "invalidText": "",
              "labelText": "First",
              "onChange": [Function],
              "onClick": [Function],
              "placeholder": "Enter first name",
              "type": "text",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "className": "bx--text__input",
              "disabled": false,
              "id": "register-last-name",
              "invalid": false,
              "invalidText": "",
              "labelText": "Last",
              "onChange": [Function],
              "onClick": [Function],
              "placeholder": "Enter last name",
              "type": "text",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2
            className="Register__title"
          >
            Title
          </h2>,
          <Form
            className="Register__form"
          >
            <TextInput
              className="bx--text__input"
              disabled={false}
              id="register-first-name"
              invalid={false}
              invalidText=""
              labelText="First"
              onChange={[Function]}
              onClick={[Function]}
              placeholder="Enter first name"
              type="text"
            />
            <TextInput
              className="bx--text__input"
              disabled={false}
              id="register-last-name"
              invalid={false}
              invalidText=""
              labelText="Last"
              onChange={[Function]}
              onClick={[Function]}
              placeholder="Enter last name"
              type="text"
            />
          </Form>,
        ],
        "className": "Register",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Title",
            "className": "Register__title",
          },
          "ref": null,
          "rendered": "Title",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <TextInput
                className="bx--text__input"
                disabled={false}
                id="register-first-name"
                invalid={false}
                invalidText=""
                labelText="First"
                onChange={[Function]}
                onClick={[Function]}
                placeholder="Enter first name"
                type="text"
              />,
              <TextInput
                className="bx--text__input"
                disabled={false}
                id="register-last-name"
                invalid={false}
                invalidText=""
                labelText="Last"
                onChange={[Function]}
                onClick={[Function]}
                placeholder="Enter last name"
                type="text"
              />,
            ],
            "className": "Register__form",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "className": "bx--text__input",
                "disabled": false,
                "id": "register-first-name",
                "invalid": false,
                "invalidText": "",
                "labelText": "First",
                "onChange": [Function],
                "onClick": [Function],
                "placeholder": "Enter first name",
                "type": "text",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "className": "bx--text__input",
                "disabled": false,
                "id": "register-last-name",
                "invalid": false,
                "invalidText": "",
                "labelText": "Last",
                "onChange": [Function],
                "onClick": [Function],
                "placeholder": "Enter last name",
                "type": "text",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
